mne/viz/evoked.py

```diff
@@ -875,3 +880,158 @@ def plot_snr_estimate(evoked, inv, show=True):
+        bad channels are excluded.
+    show : bool
+        Show figure if True.
+    ts_args : dict
+        A dict of `kwargs` that are forwarded to `evoked.plot` to
+        style the butterfly plot. `axes` and `show` are ignored.
+    topomap_args : dict
+        A dict of `kwargs` that are forwarded to `evoked.plot_topomap`
+        to style the topomaps. `axes` and `show` are ignored. If `times`
+        is not in this dict, automatic peak detection is used.
+
+    Returns
+    -------
+    fig : instance of matplotlib.figure.Figure | list
+        The figure object containing the plot. If `evoked` has multiple
+        channel types, a list of figures, one for each channel type, is
+        returned.
+
+    Notes
+    -----
+    .. versionadded:: 0.12.0
+    """
+    import matplotlib.pyplot as plt
+
+    # channel selection
+    # simply create a new evoked object(s) with the desired channel selection
+    if exclude == "bads":
+        exclude = [ch for ch in evoked.info['bads']
+                   if ch in evoked.info['ch_names']]
+    evoked = evoked.copy().drop_channels(exclude)
+    if picks is not None:
+        pick_names = [evoked.info["ch_names"][pick] for pick in picks]
+        evoked = evoked.pick_channels(pick_names)
+    info = evoked.info
+    data_types = ['eeg', 'grad', 'mag', 'seeg']
+    ch_types = set([channel_type(info, idx)
+                    for idx in range(info['nchan'])
+                    if channel_type(info, idx) in data_types])
+    # if multiple sensor types: one plot per channel type, recursive call
+    if len(ch_types) > 1:
+        print(ch_types)
+        figs = list()
+        for t in ch_types:  # pick only the corresponding channel type
+            ev_ = evoked.pick_channels([info["ch_names"][idx]
+                                        for idx in range(info['nchan'])
+                                        if channel_type(info, idx) == t],
+                                       copy=True)
+            if len(set([channel_type(ev_.info, idx)
+                        for idx in range(ev_.info['nchan'])
+                        if channel_type(ev_.info, idx) in data_types])) > 1:
+                raise RuntimeError("Possibly infinite loop due to channel "
+                                   "selection problem. This should never "
+                                   "happen! Please check your channel types.")
+            figs.append(joint_plot(ev_, title=title, show=show,
+                                   ts_args=ts_args,
+                                   topomap_args=topomap_args))
+        return figs
+
+    fig = plt.figure()
+
+    # set up time points to show topomaps for
+    if isinstance(times, string_types):
+        if times == "peaks":
+            times = _find_peaks(evoked, topomap_args.get("peaks", 3))
+        elif times == "auto":
+            times = np.linspace(evoked.times[0], evoked.times[-1], 5)
+    elif np.isscalar(times):
+        times = [times]
```

Comment:
> This probably belongs in mne.evoked, since it only applies to evoked objects, but not exclusively to viz?

Is it only used by viz functions? By evoked viz functions? If so I'd put it in the mne/viz/evoked.py

> Also I'm pretty sure you wrote the part in plot_topomap I'd be changing here (the peak finder).

That's certainly possible, I don't remember. There presumably wasn't a need to have this logic go in its own function before since (I guess) it was only used in one place, but now it makes sense to separate it out to keep the code DRY.